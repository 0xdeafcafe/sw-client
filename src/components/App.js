import React from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

const App = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
