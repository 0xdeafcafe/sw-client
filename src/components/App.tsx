
import React, {ReactNode} from 'react';
import { Container, Navbar, NavbarBrand } from 'reactstrap';

interface AppProps {
	children: ReactNode;
  }

const App: React.FC<AppProps> = ({ children }) => (
	<div className={'App'}>
		<Navbar
			color={'light'}
			light
		>
			<NavbarBrand href={'/'}>{'Star Wars Client'}</NavbarBrand>
		</Navbar>

		<Container className={'pt-5'}>
			{children}
		</Container>
	</div>
);

export default App;
