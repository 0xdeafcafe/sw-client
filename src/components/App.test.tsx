import React from "react";
import { Provider } from "react-redux";
import { configure, mount } from "enzyme";
import configureStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16";
import initialState from "../reducers/initial-state";
import thunk from "redux-thunk";
import App from "./App";

configure({ adapter: new Adapter() });

const mockStore = configureStore([thunk]);

configure({ adapter: new Adapter() });

const store = mockStore(initialState);

it("renders correctly when there are no items", () => {
  const wrapper = mount(
    <Provider store={store}>
      <App children={undefined} />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
