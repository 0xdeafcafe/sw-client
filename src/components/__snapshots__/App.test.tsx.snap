// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no items 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <div
      className="App"
    >
      <Navbar
        color="light"
        expand={false}
        light={true}
        tag="nav"
      >
        <nav
          className="navbar navbar-light bg-light"
        >
          <NavbarBrand
            href="/"
            tag="a"
          >
            <a
              className="navbar-brand"
              href="/"
            >
              Star Wars Client
            </a>
          </NavbarBrand>
        </nav>
      </Navbar>
      <Container
        className="pt-5"
        tag="div"
      >
        <div
          className="pt-5 container"
        />
      </Container>
    </div>
  </App>
</Provider>
`;
